packages/grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -561,7 +563,10 @@ export const KeyboardNavigationMixin = (superClass) =>
 
       index += step;
       while (index >= 0 && index <= tabOrder.length - 1) {
-        const rowElement = this.__rowFocusMode ? tabOrder[index] : tabOrder[index].parentNode;
+        let rowElement = tabOrder[index];
+        if (rowElement && !this.__rowFocusMode) {
+          rowElement = tabOrder[index].parentNode;
+        }
```

Comment:
Change addressed by ["should be possible to exit an empty grid with tab" test](https://github.com/vaadin/web-components/pull/2998/files#diff-9b10654eaa2484f223d9633beb33d103c30e840030fe93b4ea565a276d1c8a7dR2090)